### ISSUE TITLE
LOOOOOOONNNNG login times with system-monitor enabled - Ubuntu

### ISSUE THREAD

**Issue by BloodyIron** [paradoxxxzero/gnome-shell-system-monitor-applet#515]
I'm running gnome-shell 3.30.2 with Ubuntu 18.10

Whenever I have this extension enabled, randomly my login will take FOREVE. Or, more specifically, anywhere from like 15 seconds to 1.5 minutes+.

I have a beefy rig, ryzen 5 2600 + Samsung 970 Evo Plus, so this kind of a delay is not okay.

Not sure where to begin giving info on my setup here, since I'm new to gnome-plugins. But the issue for sure goes away when I disable just this one extension.

I'd like to help to get this sorted, because I quite like this extension otherwise.

**Comment by piegamesde** [comment]
I did experience similar login times, but I did not link them to this extension. I can't test it myself currently, because it is broken on 3.32.

> But the issue for sure goes away when I disable just this one extension.

Try the opposite: Disable all extensions and enable only this one. Can you still reproduce the issue? If no, it may be another extension or multiple ones interacting in a bad way.

**Comment by BloodyIron** [comment]
I actually went through previously and toggled them all off then back on individually to try and reproduce the issue. The issue didn't show up until I turned system-monitor on. So I'm confident that it is the cause.

**Comment by BloodyIron** [comment]
So is this abandoned? Feeling like it...

**Comment by chrisspen** [comment]
This is a duplicate of issue #511, which has had a fix merged. Assuming people are happy with master, I'll probably push up an official release in the next week or so.

**Comment by BloodyIron** [comment]
@chrisspen how is this a duplicate of support for 3.32 if the issue I'm reporting was on an earlier version of gnome?

**Comment by chrisspen** [comment]
@BloodyIron The problem effects other versions than just 3.32...

**Comment by BloodyIron** [comment]
So then why is this closed? The "dupe" ticket is about version support, not performance...

**Comment by RavanH** [comment]
I agree, this should not be closed. I am seeing this behavior on Ubuntu 19.04 on Xorg session but with **logout**, not with login. Very weird... the problem disappears completely (logout time dropping from over 40 seconds to less than 4) when system-monitor is disabled.

Wayland session does not suffer this issue except right after having run an Xorg session (with loooong logout time) and then starting a Wayland session. Then I get a looooong **login** time: more than one minute where it usually is less than 10 seconds to a full and responsive desktop. Even more weird...

**Comment by chrisspen** [comment]
@RavanH Which version of the extension are you using?

**Comment by RavanH** [comment]
@chrisspen Where do I find the version?

**Comment by chrisspen** [comment]
@RavanH Did you install from the extensions.gnome.org website or from github?

**Comment by RavanH** [comment]
The one from extensions.gnome.org does not work on Ubuntu 19.04 so I've replaced the files with the ones form Github :)

**Comment by chrisspen** [comment]
That's what I was getting at thanks.

**Comment by BloodyIron** [comment]
I'd love to retest, but 3.32 is not on the extensions page and I'm on 19.04 (Ubuntu) which is gnome-session 3.32.

**Comment by chrisspen** [comment]
@BloodyIron You'd have to install from github as explained in the readme. By the time it's on the extensions page, it's too late to test.

**Comment by RavanH** [comment]
Hi @chrisspen I've been running the extension on an Xorg session a while now and it's a very elusive issue. I noticed that the long logout time usually occurs when I have just de/activated the system-monitor extension. Also happend a few times on login but then there is also an error message from my hplip-gui about a missing system tray (it wants to add a HP logo with an options menu there) ... so maybe this is related? Or maybe just a secondary result of the long login time, not sure.

**Comment by RossLYoung** [comment]
I was experiencing this as well. I had no idea the login delay was caused by this extension, so a huge thanks for pointing it out.
I also couldn't start the system monitor from the menu #474 
When I changed my system monitor from the snap version to the apt version, the login delay has gone.
https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/issues/474#issuecomment-471163830
It has only been 48 hours, but there's been no login delay so far.
